core/node/base_token_fetcher/src/lib.rs

```diff
@@ -55,23 +60,25 @@ impl BaseTokenFetcher {
     pub async fn new(config: BaseTokenFetcherConfig) -> anyhow::Result<Self> {
         let http_client = reqwest::Client::new();
 
-        let conversion_rate = http_client
+        let conversion_rate_str = http_client
             .get(format!(
                 "{}/conversion_rate/0x{}",
                 config.host,
                 config.token_address.encode_hex::<String>()
             ))
             .send()
             .await?
-            .json::<u64>()
+            .json::<String>()
             .await
             .context("Unable to parse the response of the native token conversion rate server")?;
+        let conversion_rate = BigDecimal::from_str(&conversion_rate_str)
+            .context("Unable to parse the response of the native token conversion rate server")?;
```

Comment:
Nit:
Maybe we could also return what was the response from the server,
it can be annoying to see this error and don't know that the server is responding.